torchmultimodal/models/video_vqvae.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from torch import nn, Tensor
+from torchmultimodal.modules.layers.attention import AxialAttentionBlock
+
+from torchmultimodal.modules.layers.conv import SamePadConv3d
+
+
+class AttentionResidualBlock(nn.Module):
+    """Residual block with axial attention as designed by VideoGPT (Yan et al. 2021)
+    and used in MUGEN (Hayes et al. 2022). Follows implementation by VideoGPT:
+    https://github.com/wilson1yan/VideoGPT/blob/master/videogpt/vqvae.py
+
+    Args:
+        hidden_dim (int): size of channel dim of input. Also determines dim of linear
+                          projections Wq, Wk, and Wv in attention
+        n_head (int): number of heads in multihead attention. Must divide into hidden_dim
+                      evenly. Default is 2 from VideoGPT.
+
+    Inputs:
+        x (Tensor): a [b, c, d1, ..., dn] tensor
+    """
+
+    def __init__(self, hidden_dim: int, n_head: int = 2) -> None:
+        super().__init__()
+        # To avoid hidden dim becoming 0 in middle layers
+        if hidden_dim < 2:
+            raise ValueError("hidden dim must be at least 2")
+
+        self.block = nn.Sequential(
+            nn.BatchNorm3d(hidden_dim),
+            nn.ReLU(),
+            SamePadConv3d(hidden_dim, hidden_dim // 2, 3, bias=False),
```

Comment:
I know the downsampling comes from the VideoVQVAE implementation, but wondering if there is any case users would want to downsample with a different frequency within the attention residual block

Reply:
I am not aware of any use case of downsampling differently within the block. I am leaning towards not having this be configurable because 1) it is downsampled down and back up within the block, 2) downsampling should occur in the conv layers before this block, 3) this block is already video vqvae specific so I don't see the point of making it configurable. WDYT?